src/rgw/rgw_kms.cc

```diff
@@ -13,13 +13,91 @@
 #include "rgw/rgw_b64.h"
 #include "rgw/rgw_kms.h"
 #include "rgw/rgw_kmip_client.h"
+#include <rapidjson/allocators.h>
+#include <rapidjson/document.h>
+#include <rapidjson/writer.h>
+#include "rapidjson/error/error.h"
+#include "rapidjson/error/en.h"
 
 #define dout_context g_ceph_context
 #define dout_subsys ceph_subsys_rgw
 
 using namespace rgw;
 
 
+/**
+ * Memory pool for use with rapidjson.  This version
+ * carefully zeros out all memory before returning it to
+ * the system.
+ */
+#define ALIGNTYPE double
```

Comment:
there are some standards-track pool allocators, but...

Reply:
The framework is a requirement of rapidjson; any other implementation would require the same code overhead.  The actual allocator here can take advantage of how it's being used to avoid tracking the lengths of all the individual pieces, which is a win because many of these pieces will be very small.  Also note: no need for multiple-thread support.  General purpose allocators can't know that, so they have to have more code & overhead.  Most of the allocators I've found handle problems I don't care about, but fail to give me the hooks I want.  I suspect the glue code for such an allocator into this framework would be much larger than actual allocator here.